core/arch/arm/plat-imx/conf.mk

```diff
@@ -1,11 +1,20 @@
 PLATFORM_FLAVOR ?= mx6ulevk
 
-ifeq ($(PLATFORM_FLAVOR),mx6ulevk)
-arm32-platform-cpuarch		:= cortex-a7
+# Get SoC associated with the PLATFORM_FLAVOR
+ifeq ($(PLATFORM_FLAVOR),$(filter $(PLATFORM_FLAVOR),mx6ulevk))
+$(call force,CFG_MX6UL,y)
 endif
-ifeq ($(PLATFORM_FLAVOR),$(filter $(PLATFORM_FLAVOR),mx6qsabrelite mx6qsabresd mx6dlsabresd))
-arm32-platform-cpuarch		:= cortex-a9
+
+ifeq ($(PLATFORM_FLAVOR),$(filter $(PLATFORM_FLAVOR),mx6qsabrelite mx6qsabresd))
```

Comment:
@jforissier, I don't agree. For maintenance, it is easier to explicitly list the target platforms.
Or use something like this:
```
ifeq ($(PLATFORM_FLAVOR),mx6ulevk)
...
else ifeq ($(PLATFORM_FLAVOR),mx6dlsabresd)
...
else # assume it is mx6qsabrelite, mx6qsabresd or mx6dlsabresd
...
endif
```
But I don't like it neither as an invalid PLATFORM_FLAVOR will select a target instead of complaining.

Maybe something like this:
```
mx6ul-flavorlist = mx6ulevk
mx6dl-flavorlist = mx6dlsabresd
mx6q-flavorlist = mx6qsabrelite mx6qsabresd

ifeq ($(PLATFORM_FLAVOR),$(filter $(PLATFORM_FLAVOR),$(mx6ul-flavorlist)))
$(call force,CFG_MX6UL,y)
else ifeq ($(PLATFORM_FLAVOR),$(filter $(PLATFORM_FLAVOR),$(mx6dl-flavorlist)))
$(call force,CFG_MX6DL,y)
else ifeq ($(PLATFORM_FLAVOR),$(filter $(PLATFORM_FLAVOR),$(mx6q-flavorlist)))
$(call force,CFG_MX6Q,y)
else
$(error Unsupported PLATFORM_FLAVOR "$(PLATFORM_FLAVOR)")
endif
```